src/utilities.ts

```diff
@@ -272,3 +272,10 @@ export function frame(buffer, frameLength, hopLength) {
     .fill(0)
     .map((_, i) => buffer.slice(i * hopLength, i * hopLength + frameLength));
 }
+
+export function magnitudeForComplexSpectrum(complexSpectrum) {
+  return complexSpectrum.real.map((real, i) => {
+    const imag = complexSpectrum.imag[i];
+    return Math.sqrt(Math.pow(real, 2) + Math.pow(imag, 2));
+  });
+}
```

Comment:
Don't we have this somewhere already? Many feature extractors already use the amplitude spectrum AFAIK.

Reply:
Yes - it's [here](https://github.com/meyda/meyda/blob/main/src/main.ts#L425) - and even in this PR we actually use that implementation, not this one. This function is from before I remembered that magnitude spectrum and amplitude spectrum are the same.

In that part of the source, the calculation is an inline snippet in a function for preparing the spectrum, and it only calculates the lower half of the buffer (which is good). But when I swapped it out for half the result of this function, I got a different result, so I wanted to keep this unused function around and debug it later.